File: backoffice/src/Service/ApiService.php
<?php

namespace App\Service;

use Symfony\Contracts\HttpClient\HttpClientInterface;

class ApiService
{
    private string $baseUrl;

    public function __construct(
        private HttpClientInterface $httpClient,
        string $apiBaseUrl = 'http://localhost:8000'
    ) {
        $this->baseUrl = rtrim($apiBaseUrl, '/');
    }

    // ========================
    // SOIGNANTS
    // ========================

    public function getSoignants(): array
    {
        return $this->get('/api/soignants');
    }

    public function getSoignant(string $matricule): array
    {
        return $this->get('/api/soignants/' . $matricule);
    }

    public function createSoignant(array $data): array
    {
        return $this->post('/api/soignants', $data);
    }

    public function updateSoignant(string $matricule, array $data): array
    {
        return $this->put('/api/soignants/' . $matricule, $data);
    }

    public function deleteSoignant(string $matricule): void
    {
        $this->delete('/api/soignants/' . $matricule);
    }

    // Specialisations soignant <-> espece
    public function getSoignantEspeces(string $matricule): array
    {
        return $this->get('/api/soignants/' . $matricule . '/especes');
    }

    public function affecterEspece(string $matricule, int $especeId): array
    {
        return $this->post('/api/soignants/' . $matricule . '/especes', [
            'id_espece' => $especeId,
        ]);
    }

    public function retirerEspece(string $matricule, int $especeId): void
    {
        $this->delete('/api/soignants/' . $matricule . '/especes/' . $especeId);
    }

    // ========================
    // ESPECES
    // ========================

    public function getEspeces(): array
    {
        return $this->get('/api/especes');
    }

    public function getEspeceAnimaux(int $especeId): array
    {
        return $this->get('/api/especes/' . $especeId . '/animaux');
    }

    public function getEspeceMenus(int $especeId): array
    {
        return $this->get('/api/especes/' . $especeId . '/menus');
    }

    public function recommanderMenu(int $especeId, int $menuId): array
    {
        return $this->post('/api/especes/' . $especeId . '/menus', [
            'id_menu' => $menuId,
        ]);
    }

    public function retirerMenu(int $especeId, int $menuId): void
    {
        $this->delete('/api/especes/' . $especeId . '/menus/' . $menuId);
    }

    // ========================
    // MENUS
    // ========================

    public function getMenus(): array
    {
        return $this->get('/api/menus');
    }

    public function getMenu(int $id): array
    {
        return $this->get('/api/menus/' . $id);
    }

    public function createMenu(array $data): array
    {
        return $this->post('/api/menus', $data);
    }

    public function updateMenu(int $id, array $data): array
    {
        return $this->put('/api/menus/' . $id, $data);
    }

    public function deleteMenu(int $id): void
    {
        $this->delete('/api/menus/' . $id);
    }

    // ========================
    // GESTIONNAIRES
    // ========================

    public function loginGestionnaire(string $login, string $password): array
    {
        return $this->post('/api/gestionnaires/login', [
            'login' => $login,
            'mot_de_passe' => $password,
        ]);
    }

    public function getGestionnaires(): array
    {
        return $this->get('/api/gestionnaires');
    }

    public function createGestionnaire(array $data): array
    {
        return $this->post('/api/gestionnaires', $data);
    }

    public function updateGestionnaire(int $id, array $data): array
    {
        return $this->put('/api/gestionnaires/' . $id, $data);
    }

    public function deleteGestionnaire(int $id): void
    {
        $this->delete('/api/gestionnaires/' . $id);
    }

    // ========================
    // HTTP helpers
    // ========================

    private function get(string $uri): array
    {
        $response = $this->httpClient->request('GET', $this->baseUrl . $uri);
        return $response->toArray();
    }

    private function post(string $uri, array $data): array
    {
        $response = $this->httpClient->request('POST', $this->baseUrl . $uri, [
            'json' => $data,
        ]);
        return $response->toArray();
    }

    private function put(string $uri, array $data): array
    {
        $response = $this->httpClient->request('PUT', $this->baseUrl . $uri, [
            'json' => $data,
        ]);
        return $response->toArray();
    }

    private function delete(string $uri): void
    {
        $this->httpClient->request('DELETE', $this->baseUrl . $uri);
    }
}

File: mobile/app/src/main/java/com/sigansud/app/api/ApiService.java
package com.sigansud.app.api;

import com.sigansud.app.models.Animal;
import com.sigansud.app.models.Espece;
import com.sigansud.app.models.LoginResponse;
import com.sigansud.app.models.Menu;
import com.sigansud.app.models.Repas;
import com.sigansud.app.models.Soignant;

import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface ApiService {

    // ─── AUTHENTIFICATION ────────────────────────────────────────────────────
    @POST("login")
    Call<LoginResponse> login(@Body Map<String, String> credentials);

    // ─── ESPÈCES ──────────────────────────────────────────────────────────────
    @GET("especes")
    Call<List<Espece>> getEspeces();

    // ─── ANIMAUX par espèce ───────────────────────────────────────────────────
    @GET("especes/{idEspece}/animaux")
    Call<List<Animal>> getAnimauxByEspece(@Path("idEspece") int idEspece);

    // ─── MENUS ────────────────────────────────────────────────────────────────
    @GET("menus")
    Call<List<Menu>> getMenus();

    @GET("especes/{idEspece}/menus")
    Call<List<Menu>> getMenusByEspece(@Path("idEspece") int idEspece);

    // ─── REPAS ────────────────────────────────────────────────────────────────
    @POST("repas")
    Call<Map<String, Object>> createRepas(@Body Map<String, Object> body);

    @GET("especes/{idEspece}/animaux/{nomBapteme}/repas")
    Call<List<Repas>> getRepasByAnimal(
            @Path("idEspece") int idEspece,
            @Path("nomBapteme") String nomBapteme
    );

    // ─── SOIGNANTS ────────────────────────────────────────────────────────────
    @GET("soignants/{matricule}")
    Call<Soignant> getSoignant(@Path("matricule") String matricule);

    @GET("soignants/{matricule}/especes")
    Call<List<Espece>> getEspecesSoignant(@Path("matricule") String matricule);
}
